codalab/lib/interactive_session.py

```diff
@@ -142,30 +143,34 @@ def get_docker_path(sub_path):
             else:
                 volumes[get_docker_path(key)] = dependency_local_path
 
-        name = self._get_container_name()
-        # Start a container as a root user. Root (id = 0) is the default user within a container.
-        command = [
+        name: str = self._get_container_name()
+        container_working_directory: str = f'{os.path.sep}{self._session_uuid}'
+
+        # Start a container as a non-root user
+        command: List[str] = [
             'docker run',
             '-it',
             f'--name {name}',
-            f'-w {os.path.sep}{self._session_uuid}',
-            # TODO: figure out a way to run as a non-root user and be able to save the bash_history on any machine
-            # '-u 1',
+            f'-w {container_working_directory}',
+            f'-e HOME={container_working_directory}',
+            f'-e HISTFILE={InteractiveSession._BASH_HISTORY_CONTAINER_PATH}',
+            '-e PROMPT_COMMAND="history -a"',
+            '-u $(id -u):$(id -g)',
         ]
         command.extend(
             [
                 # Example: -v local_path/some_folder:/0x707e903500e54bcf9b072ac7e3f5ed36_dependencies/foo:ro
-                '-v {}:{}:ro'.format(local_path, docker_path)
+                f'-v {local_path}:{docker_path}:ro'
                 for docker_path, local_path in volumes.items()
             ]
         )
-        command.append(
-            '-v {}:{}:rw'.format(
-                self._host_bash_history_path, InteractiveSession._BASH_HISTORY_CONTAINER_PATH
-            )
+        command.extend(
+            [
+                f'-v {self._host_bash_history_path}:{InteractiveSession._BASH_HISTORY_CONTAINER_PATH}:rw',
+                f'-v {self._host_bundle_path}:{container_working_directory}:rw',
```

Comment:
Before the root user had access to the working directory without mounting. Since we are running as a non-root user, we have to mount this directory from the host so the non-root user has access in the container.